packages/dds/matrix/src/test/memory/sharedMatrix.spec.ts

```diff
@@ -0,0 +1,205 @@
+/*!
+ * Copyright (c) Microsoft Corporation and contributors. All rights reserved.
+ * Licensed under the MIT License.
+ */
+
+import { type IMemoryTestObject, benchmarkMemory } from "@fluid-tools/benchmark";
+import { MockFluidDataStoreRuntime } from "@fluidframework/test-runtime-utils/internal";
+
+import { SharedMatrix } from "../../index.js";
+import { matrixFactory, createLocalMatrix } from "../utils.js";
+
+describe("SharedMatrix memory usage", () => {
+	const matrixValue = "testValue";
+	const matrixSize = [10, 100, 1000, 10000];
+	const operationCounts = [10, 100, 1000, 10000];
+
+	beforeEach(async () => {});
+
+	afterEach(() => {});
+
+	for (const size of matrixSize) {
+		describe(`Size of ${size}*${size} SharedMatrix`, () => {
+			for (const count of operationCounts) {
+				benchmarkMemory(
+					new (class implements IMemoryTestObject {
+						readonly title = `Insert a column in the middle ${count} times`;
+						private localMatrix: SharedMatrix = matrixFactory.create(
+							new MockFluidDataStoreRuntime(),
+							"testLocalMatrix",
+						);
+
+						async run(): Promise<void> {
+							for (let i = 0; i < count; i++) {
+								this.localMatrix.insertCols(size / 2, 1);
+							}
+						}
+
+						beforeIteration(): void {
+							this.localMatrix = createLocalMatrix("testLocalMatrix", size, matrixValue);
+						}
+					})(),
+				);
+
+				benchmarkMemory(
```

Comment:
For all of the removal cases, we probably want to skip any matrix-size / operation-count combinations where combination-count \>= matrix-size. That these tests are passing seems to indicate that SharedMatrix is implemented in such a way where removing a non-existent row/column is a no-op, but we probably shouldn't depend on that. Also, the SharedTree code will error in such a case.

Reply:
change to removeRow/Cols(0,1)

Reply:
That doesn't resolve the underlying problem, though. If we have an initial table with 10 rows, and we are removing 100 rows, after the 10th iteration there will be no rows left to remove. That may not cause SharedMatrix to crash (though it probably should), but it isn't useful to test. 

I would recommend putting the removal test cases in an if check so that they only run if `operationCount <= `matrixSize`.

Reply:
And then you should be able to make the removal occur in the middle again.

Reply:
Ah! I got it now. Make a. valid count array